Initial DoCM Importer

acoffman opened the pull request:
This is DoCM importer that @bainscou and I worked on. It is working successfully and imports the following items:

````
30 gene_claims.
30 gene_claim_aliases.
35 drug_claims.
88 interaction_claims.
176 interaction_claim_attributes.
````

All Genes have an Entrez ID and are thus grouped correctly by the gene grouper. Before merging we did have two questions:

 * There is a column in the DoCM TSV called `effect` which contains values like 
```
decreased sensitivity
effect
no response
educed sensitivity
resistance
response
sensitivity
````

and it is unclear to me if that should be imported as the `interaction_type` or the `known_action_type` of the interaction. 

 * Where should DoCM lie in the sort order of sources?

With those two questions answered, we should be good to go!

obigriffith commented:
Just discussed with Ben. Right now I think that column in the TSV is 
neither interaction_type nor known_action_type. It describes the nature 
of the variant's effect on the drug rather than the drug's effect on the 
gene (gene product). Therefore, it should just be meta-data (variant 
effect?) for the drug gene interaction.

Now would be a good time to revisit the overall source ranking. Ben is 
going to send around the current list. I think that CIViC and DoCM 
should appear near the top, perhaps even before MCG. There are some 
other sources (e.g., MCG Clinical trials) that are currently ranked too 
high. But, finalizing this ranking shouldn't delay this PR.

On 8/26/15 11:04 AM, Adam Coffman wrote:
>
> This is DoCM importer that @bainscou <https://github.com/bainscou> and 
> I worked on. It is working successfully and imports the following items:
>
> |30 gene_claims.
> 30 gene_claim_aliases.
> 35 drug_claims.
> 88 interaction_claims.
> 176 interaction_claim_attributes.
> |
>
> All Genes have an Entrez ID and are thus grouped correctly by the gene 
> grouper. Before merging we did have two questions:
>
>   * There is a column in the DoCM TSV called |effect| which contains
>     values like
>
> |decreased sensitivity
> effect
> no response
> educed sensitivity
> resistance
> response
> sensitivity
> |
>
> and it is unclear to me if that should be imported as the 
> |interaction_type| or the |known_action_type| of the interaction.
>
>   * Where should DoCM lie in the sort order of sources?
>
> With those two questions answered, we should be good to go!
>
> ------------------------------------------------------------------------
>
>
>         You can view, comment on, or merge this pull request online at:
>
> https://github.com/genome/dgi-db/pull/38
>
>
>         Commit Summary
>
>   * initial docm importers
>   * initial docm tsv importer
>
>
>         File Changes
>
>   * *A* lib/genome/importers/docm/docm_row.rb
>     <https://github.com/genome/dgi-db/pull/38/files#diff-0> (15)
>   * *A* lib/genome/importers/docm/docm_tsv_importer.rb
>     <https://github.com/genome/dgi-db/pull/38/files#diff-1> (32)
>   * *A* lib/genome/updaters/get_docm.rb
>     <https://github.com/genome/dgi-db/pull/38/files#diff-2> (78)
>
>
>         Patch Links:
>
>   * https://github.com/genome/dgi-db/pull/38.patch
>   * https://github.com/genome/dgi-db/pull/38.diff
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/genome/dgi-db/pull/38>.
>

-- 
Obi L. Griffith, PhD
Assistant Professor of Medicine, Division of Oncology
Assistant Director, McDonnell Genome Institute
Washington University School of Medicine
Campus Box 8501, Room 5140
4444 Forest Park Ave
St. Louis, MO, 63108
Phone: 314-747-9248
www.obigriffith.org



____
This email message is a private communication. The information transmitted, including attachments, is intended only for the person or entity to which it is addressed and may contain confidential, privileged, and/or proprietary material. Any review, duplication, retransmission, distribution, or other use of, or taking of any action in reliance upon, this information by persons or entities other than the intended recipient is unauthorized by the sender and is prohibited. If you have received this message in error, please contact the sender immediately by return email and delete the original message from all computer systems. Thank you.

acoffman commented:
Added the effect column as metadata and pushed. I'll go ahead and merge this then. Once all the new and updated importers are merged we'll want to make a new data snapshot to push out to the public site.
